### PR TITLE
4/editing users

### PR DESCRIPTION
**Pull request by denis-onder**
Issue #4 

TL;DR - What does this implement?
---------------------------------
Mailing system, password recovery, caching, account, email change and password change verification.

How to Test
-----------
Start the Node server using `npm start`,
then run `npm test`.

Worries / Concerns
------------------
Again, security.
I just remembered one weak point of JWT authentication that I didn't list as a concern in my previous PR. If a user changes it's credentials, removes it's account or changes the account in question in any way, the token will still be valid for use until it expires for some endpoints. The next thing to be worked on should most certainly be switching over to OAuth. 

EDIT: Another concern is caching. Would it be better to run a Redis instance for caching, or should we keep running the current cache setup?

Checklist
---------

 - [ ] Do we need to test this?
 - [X] Were tests written?
 - [X] Does this meet coding standards?
 - [X] Is this is code change?
 - [ ] Is this a meta change (e.g issue template)?
 - [X] Is this pull request ready for review?
 - [ ] Is this a breaking change (e.g not backwards compatible with clients)?
 
ESLint Output
-------------

```
[Info  - 5:48:26 PM] ESLint server stopped.
[Info  - 5:48:26 PM] ESLint server running in node v10.11.0
[Info  - 5:48:26 PM] ESLint server is running.
[Info  - 5:48:27 PM] ESLint library loaded from: /home/denis/code/nacho-night-auth-api/node_modules/eslint/lib/api.js
Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting
```

Reviewers
---------
@gmemstr

### CODE REVIEW

File: app/validation/index.js [hunk 1]
```diff
@@ -43,12 +43,46 @@ class Validator {
...
+    if (!isEmail(data.email)) {
+      this.errors.emailInvalid = 'Please provide a valid email address.';
+    }
+    if (!isValid(data.password)) {
+      this.errors.passwordInputInvalid = 'The provided input for the password is not valid.';
+    }
+    if (isEmpty(data.password)) {
+      this.errors.passwordEmpty = 'Please provide your password.';
+    }
+    return this.checkForErrors();
```
Comment (autumnation) [inline app/validation/index.js]: I think it would be best to have one error field, then progressively check for the errors (i.e. first email, then valid, then password input, then password) so it looks cleaner.

File: app/controller.js [hunk 2]
```diff
@@ -75,6 +96,101 @@ class Controller {
...
+  }
+
+  async recover(req, res) {
+    // Recover the password
+    try {
+      const email = cache.get(req.params.token);
+      if (email === undefined) return res.status(500).json({ error: 'An error has occured.' });
+      const user = await User.findOne({ where: { email } });
+      if (!user) return res.status(404).json({ error: 'This email address is not in use.' });
+      await user.update({ password: hashSync(req.body.password, 10) });
```
Comment (gmemstr) [inline app/controller.js]: I wonder if we should do more rounds for the password. Standard practice right now seems to be 12 or 14.

**Comment by denis-onder** [comment]
Changed numbers of salt rounds to 14, refactored the validator class and updated docs.
